onnxruntime/core/providers/cpu/nn/tfidfvectorizer.cc

```diff
@@ -455,28 +455,33 @@ Status TfIdfVectorizer::ComputeImpl(OpKernelContext* ctx) const {
     C = input_dims[0];
     if (C < 1) {
```

Comment:
>(C < 1)  [](start = 7, length = 8)

Should it be < 0?

Reply:
I did not change it because I did not need to. In my case, I distinguish between empty input with shape {0} and empty input with shape {B, 0} with B > 0. Shape {B,0} with B > 0 is produced by the previous operator when it cannot compute any output for every input observation but it keeps the number of processed observations. The first one (shape={0}) means there was no data to begin with, the first dimension usually denotes the number of observations to process. Shape {0} could be unexpectdly produced by a previous operator if it removes observations (like a filter). I think if we allow this case, we might have to allow it in every operator.

Reply:
The code is not readable now, plus, frequencies are allocated using artificial dimension of b_dim = 1 in that case. So you want to redo the code in a different way.

---
In reply to: [268783844](https://github.com/Microsoft/onnxruntime/pull/702#discussion_r268783844) [](ancestors = 268783844)

Reply:
@Xavier, I feel supporting input shape [0] is reasonable. What if user just have to have a vector/count representation of a single document? I guess [0] is a valid possibility in that case.

---
In reply to: [268789799](https://github.com/Microsoft/onnxruntime/pull/702#discussion_r268789799) [](ancestors = 268789799,268783844)

Reply:
You can see the current behaviour in the unit test (look for substring empty).